apps/web/test/lib/getSchedule.test.ts

```diff
@@ -1225,6 +1225,89 @@ describe("getSchedule", () => {
         }
       );
     });
+
+    test("check the availability of guest when rescheduling if guest is a cal.com user", async () => {
+      const { dateString: plus1DateString } = getDate({ dateIncrement: 1 });
+      const { dateString: plus2DateString } = getDate({ dateIncrement: 2 });
+
+      await createBookingScenario({
+        eventTypes: [
+          {
+            id: 1,
+            slotInterval: 60,
+            length: 60,
+            users: [{ id: 101 }],
+          },
+          {
+            id: 2,
+            slotInterval: 60,
+            length: 60,
+            users: [{ id: 102 }],
+          },
+        ],
+        users: [
+          {
+            ...TestData.users.example,
+            id: 101,
+            schedules: [TestData.schedules.IstWorkHours],
+            email: "user1@example.com",
+          },
+          {
+            ...TestData.users.example,
+            id: 102,
+            schedules: [TestData.schedules.IstWorkHours],
+            email: "user2@example.com",
+          },
+        ],
+        bookings: [
+          {
+            userId: 101,
+            eventTypeId: 1,
+            uid: "uid1",
+            status: "ACCEPTED",
+            startTime: `${plus1DateString}T04:30:00.000Z`,
+            endTime: `${plus1DateString}T05:30:00.000Z`,
+            attendees: [{ email: "user2@example.com" }],
+          },
+          {
+            userId: 102,
+            eventTypeId: 2,
+            uid: "uid2",
+            status: "ACCEPTED",
+            startTime: `${plus2DateString}T06:30:00.000Z`,
+            endTime: `${plus2DateString}T07:30:00.000Z`,
+          },
+        ],
+      });
+
+      const scheduleForTeamEventOnADayWithNoBooking = await getSchedule({
+        input: {
+          eventTypeId: 1,
+          eventTypeSlug: "",
+          startTime: `${plus1DateString}T18:30:00.000Z`,
+          endTime: `${plus2DateString}T18:29:59.999Z`,
+          timeZone: Timezones["+5:30"],
+          isTeamEvent: false,
+          rescheduleUid: "uid1",
+        },
+      });
+
+      expect(scheduleForTeamEventOnADayWithNoBooking).toHaveTimeSlots(
+        [
+          `04:30:00.000Z`,
+          `05:30:00.000Z`,
+          // `06:30:00.000Z`, // <- This slot should be occupied by the Pending Requires Confirmation booking blocking this slot
+          `07:30:00.000Z`,
+          `08:30:00.000Z`,
+          `09:30:00.000Z`,
+          `10:30:00.000Z`,
+          `11:30:00.000Z`,
+        ],
+        {
+          dateString: plus2DateString,
+        }
+      );
+    });
```

Comment:
There seems to be a bug in prismock which adds both the users (`id: 101` and `id: 102`) to the users field in both the event types even though event 1 (`id: 1`) only has user 101 (`id: 101`) in the users array and the same goes for event 2 (`id: 2`). You can log the `allEventTypes` variable [here](https://github.com/calcom/cal.com/blob/main/apps/web/test/utils/bookingScenario/bookingScenario.ts#L249) and run the test and notice that the events have both the users added in the users field.
Because of that this test passes even when the availability of attendees is not being checked.

Reply:
It shows outdated here, has this been fixed @kart1ka ?

Reply:
The issue still persists. Seems to be some kind of a prismock bug. I have been trying to fix it, but no luck so far.

Reply:
Hi @Amit91848 

The issue seems to come from how Prismock handles the implicit many-to-many relationship between User and EventType model (`user_eventtype`). In the Prisma data model, `user` and `eventType` columns do not have the `relationFromFields` and `relationToFields` mappings, so those arrays are left empty.

Because of that, Prismock can’t define the right `where` clause to get the correct user for each event type and ends up fetching all users in the User table instead.

And because it fetches all the users for the eventType, the test ends up passing even when we do not specifically check for the attendee's availability because the attendee gets included when we select the `users` field while fetching the eventType [here](https://github.com/calcom/cal.com/blob/2146c0444421cb8b07061ba8b5a4235003ef1fbd/packages/trpc/server/routers/viewer/slots/util.ts#L230).

I'm not sure what could be a workaround for this issue.